ThumbnailSequence.cpp

```diff
@@ -1560,7 +1560,7 @@ ThumbnailSequence::CompositeItem::CompositeItem(
 bool
 ThumbnailSequence::CompositeItem::incompleteThumbnail() const
 {
-	return dynamic_cast<IncompleteThumbnail*>(m_pThumb) != 0;
+	return m_pThumb == 0 && dynamic_cast<IncompleteThumbnail*>(m_pThumb) != 0;
```

Comment:
I already recompiled qt and tried gcc 4.8 and 4.9. I remember from a former c++ project that freebsd revealed real code errors on which linux didn't crash.  Maybe it is similar here.
I cite my stackoverflow link:
"Let me add to this that you should, by all means, replace auto_ptr by unique_ptr in your code and fix the compilation errors. You'd be surprised how many bugs this will uncover."
Maybe filters/page_split/Task.cpp line 208 is such a case. Maybe you can suggest a solution to that problem and I can fix the other compilation errors?
```
filters/page_split/Task.cpp:213:4: error: use of deleted function 'std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = DebugImagesImpl; _Dp = std::default_delete<DebugImagesImpl>]'
```
(Copy problem) How to pass it without copying or copy the right way? Use a shared_ptr?